### ISSUE TITLE
toml file parsing error in v2.0:alpha6

### ISSUE THREAD

**Issue by dalanmiller** [containous/traefik#4987]
### Do you want to request a *feature* or report a *bug*?

Bug

### Did you try using a 1.7.x configuration for the version 2.0?

- [ ] Yes
- [x] No

### What did you do?

> dalan [Today at 1:11 PM]
> Getting an error on the new `2.0.0-alpha6`
> 
> traefik                 | 2019/06/20 13:09:38 command /traefik error: entryPoints cannot be a standalone > element (type static.EntryPoints)
> 
> This is my `traefik.toml` file
> 
> 
> [entryPoints]
>   [entrypoints.http]
>     address = ":80"
> 
>   [entrypoints.https]
>     address = ":443"
> 
> It appears the docs haven't changed for defining entrypoints, so I'm not sure what's going wrong. It'd be super useful if the line the parse error bailed out on was logged. (edited)
> 12 replies
> ldez  [11 minutes ago]
> @dalan could you try to remove the first `[entrypoints]` ? (edited)
> dalan  [9 minutes ago]
> That was it :+1:
> dalan  [9 minutes ago]
> Now getting `field not found node: domain`
> ldez  [8 minutes ago]
> give me your configuration
> dalan  [8 minutes ago]
> OOC where is the toml file being used for tests?
> ldez  [8 minutes ago]
> in the majority of the integration tests
> ldez  [5 minutes ago]
> could you give me more information about your run and configure?
> ldez  [4 minutes ago]
> docker-compose file, toml file, ...
> dalan  [4 minutes ago]
> https://github.com/containous/traefik/blob/master/integration/fixtures/file/simple.toml#L8
> dalan  [3 minutes ago]
> Shouldn't the tests have failed for `alpha6` if parsing the above? (edited)
> ldez  [2 minutes ago]
> we have a regression on file parsing, could you open an issue thanks. I will fixed that asap (edited)
> dalan  [1 minute ago]
> Can do! Thanks as always for your help @ldez :pray:

https://traefik.slack.com/archives/C0CDT22PJ/p1561036260137500

### What did you expect to see?

```
[entryPoints]
  [entrypoints.http]
    address = ":80"

  [entrypoints.https]
    address = ":443"
```

Should work fine in a config file. Plus this should've been caught via the simple toml file being parsed in the integration tests.

https://github.com/containous/traefik/blob/master/integration/fixtures/file/simple.toml#L8

### What did you see instead?
```
traefik                 | 2019/06/20 13:22:05 command /traefik error: entryPoints cannot be a standalone element (type static.EntryPoints)
```
### Output of `traefik version`: (_What version of Traefik are you using?_)

```
Version:      2.0.0-alpha6
Codename:     faisselle
Go version:   go1.12.6
Built:        2019-06-18T18:25:38Z
OS/Arch:      linux/amd64
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

n/a

**Comment by ldez** [comment]
The root of the issue is case: `entryPoints` != `entrypoints`

TOML is case sensitive, we use TOML lib that have a weird behavior this case sensitivity:

Valid:
```toml
[entrypoints]
[entrypoints.http]
address = ":80"
``` 

Valid:
```toml
[entryPoints]
[entryPoints.http]
address = ":80"
``` 
Valid:
```toml
[EntryPoints]
[EntryPoints.http]
address = ":80"
``` 

Not valid:
```toml
[entryPoints]
[entrypoints.http]
address = ":80"
```

It's not a new behavior, but we are more strict in the v2 than in the v1 about this kind of typo.

**Comment by polarathene** [comment]
@ldez That's kind of important to make a note of in the documentation then. The inconsistency in examples in the docs threw me off and gave the impression that there was no case sensitivity.

[File Provider](https://docs.traefik.io/v2.0/providers/file/):

```toml
  [providers.file]
    filename = "rules.toml"
```

[Static file reference](https://docs.traefik.io/v2.0/reference/static-configuration/file/):

```toml
  [Providers.File]
    Filename = "foobar"
```

See how there is a switch to using Pascal Case, where other parts of docs show Camel Case. Perhaps a find/replace on `EntryPoints.` to `entryPoints.` and similar would help here? 

Mentioning the importance of case consistency for file provider / toml config could be good [here](https://docs.traefik.io/v2.0/providers/file/)? (Maybe in the reference for static/dynamic too).

**Comment by dalanmiller** [comment]
Does the file example in this middleware work? Does the [pascal|lowercase|camel] casing need to be throughout the config or can individual components like `forwardAuth` in the example below each be their own casing? 

```
# Forward authentication to authserver.com
[http.middlewares]
  [http.middlewares.test-auth.forwardAuth]
    address = "https://authserver.com/auth"
    trustForwardHeader = true
    authResponseHeaders = ["X-Auth-User", "X-Secret"]

    [http.middlewares.test-auth.forwardauth.tls]
      ca = "path/to/local.crt"
      caOptional = true
      cert = "path/to/foo.cert"
      key = "path/to/foo.key"
``` 

https://docs.traefik.io/v2.0/middlewares/forwardauth/

**Comment by polarathene** [comment]
The docs have some consistency issues and I think you spotted one there(I haven't tried it as I can't seem to get tls to work). 

AFAIK, the casing type doesn't matter as long as it's consistent for any repetition downwards, so since the parent is `forwardAuth` you would need to keep that as well, but any sub/child fields like `.tls` could be `.TLS`. Likewise you could have `http.MIDDLEWARES`, so long as that casing was consistent.

**Comment by dalanmiller** [comment]
Alright, so this isn't a regression but the docs should:

1. Mention that any sort of casing is sufficient as long as it's consistent throughout your `*.toml` 
2. Be consistent across all the code snippet examples (does containous have a preference?) 
3. Should have tests for this to ensure that the chosen casing is maintained.

**Comment by polarathene** [comment]
Any suggestions for where that should be mentioned? Perhaps in the Getting Started section? I know that the docs mention the TOML format, but people want to just get started and may not read up enough to be familiar with the casing mattering in this way(as it's not immediately apparent). The docs having a consistent casing convention would at least avoid this issue pretty well. Perhaps the reference pages could also hyperlink back to the Getting Started section on TOML casing if that'd be useful too.

**Comment by geraldcroes** [comment]
Closed in favor of #5028 (we need to fix the documentation)
Thank you for helping us improving Traefik!
